ports/winit/app.rs

```diff
@@ -346,12 +346,27 @@ fn get_default_url() -> ServoUrl {
     // If the url is not provided, we fallback to the homepage in prefs,
     // or a blank page in case the homepage is not set either.
     let cwd = env::current_dir().unwrap();
-    let cmdline_url = opts::get().url.clone();
+    let mut cmdline_url = opts::get().url.clone();
     let pref_url = {
         let homepage_url = pref!(shell.homepage);
         parse_url_or_filename(&cwd, &homepage_url).ok()
     };
     let blank_url = ServoUrl::parse("about:blank").ok();
 
+    // Check if a command-line URL is provided
+    if let Some(url) = cmdline_url.clone() {
+        let url_path = url.path();
```

Comment:
Maybe you should check if the URL is relative (scheme and host are empty) before going down the path of checking the filesystem.